torch_harmonics/convolution.py

```diff
@@ -134,17 +135,18 @@ def _normalize_convolution_tensor_s2(
     return psi_vals
 
 
+@lru_cache(typed=True, copy=True)
 def _precompute_convolution_tensor_s2(
-    in_shape,
-    out_shape,
-    filter_basis,
-    grid_in="equiangular",
-    grid_out="equiangular",
-    theta_cutoff=0.01 * math.pi,
-    theta_eps = 1e-3,
-    transpose_normalization=False,
-    basis_norm_mode="mean",
-    merge_quadrature=False,
+    in_shape: Tuple[int],
+    out_shape: Tuple[int],
```

Comment:
it should also work with Lists fyi.

Reply:
that is not possible unfortunately, since List is not a haschable datatype.